cpg-neo4j/src/main/kotlin/de/fraunhofer/aisec/cpg/persistence/Common.kt

```diff
@@ -0,0 +1,244 @@
+ *
+ * This mutable map uses a Kotlin class type as the key and a set of strings representing associated
+ * labels as the value. The [labelCache] provides efficient lookup and prevents redundant
+ * re-computation of labels for the same class type.
+ */
+val labelCache: MutableMap<KClass<*>, Set<String>> = mutableMapOf()
+
+/**
+ * A cache mapping classes of type [Persistable] to their respective properties.
+ *
+ * This mutable map stores metadata about [Persistable] objects. For each specific class that
+ * implements the [Persistable] interface, it caches a mapping between property names and their
+ * corresponding [KProperty1] references. This allows efficient reflection-based access to class
+ * properties without repeatedly inspecting the class at runtime.
+ *
+ * The key in the map is the [KClass] of a subclass of [Persistable]. The value is a [Map] where the
+ * keys are strings representing the property names, and the values are [KProperty1] references
+ * pointing to those properties. This can be used for dynamic property access or serialization
+ * processes.
+ */
+val schemaPropertiesCache:
+    MutableMap<KClass<out Persistable>, Map<String, KProperty1<out Persistable, *>>> =
+    mutableMapOf()
+
+/**
+ * A set containing predefined property types represented as Kotlin type objects that can be used as
+ * properties in [schemaProperties].
+ */
+val propertyTypes =
+    setOf(
+        String::class.createType(),
+        Int::class.createType(),
+        Long::class.createType(),
+        Boolean::class.createType(),
+        Name::class.createType(),
+        Uuid::class.createType(),
+        Granularity::class.createType(),
+        DependenceType::class.createType(),
+    )
+
+internal val log = LoggerFactory.getLogger("Persistence")
+
+/**
+ * Returns the [Persistable]'s properties. This DOES NOT include relationships, but only properties
+ * directly attached to the node/edge.
+ */
+fun Persistable.properties(): Map<String, Any?> {
+    val properties = mutableMapOf<String, Any?>()
+    for (entry in this::class.schemaProperties) {
+        val value = entry.value.call(this)
+
+        if (value == null) {
+            continue
+        }
+
+        value.convert(entry.key, properties)
+    }
+
+    return properties
+}
+
+/**
+ * Runs any conversions that are necessary by [CompositeAttributeConverter] and
+ * [org.neo4j.ogm.typeconversion.AttributeConverter]. Since both of these classes are Neo4J OGM
+ * classes, we need to find new base types at some point.
+ */
+fun Any.convert(originalKey: String, properties: MutableMap<String, Any?>) {
+    // TODO: generalize conversions
+    if (this is Name && originalKey == "name") {
+        properties += NameConverter().toGraphProperties(this)
+    } else if (this is Name) {
+        properties.put(originalKey, SimpleNameConverter().toGraphProperty(this))
+    } else if (this is PhysicalLocation) {
```

Comment:
Time to integrate converter functionality into the converted class? `PhysicalLocation.toGraphProperties()`